### ISSUE TITLE
Reduce parser allocations by testing for known strings

### ISSUE THREAD

**Issue by Kaliumhexacyanoferrat** [Kaliumhexacyanoferrat/GenHTTP#475]
As an operator of a web application written with GenHTTP, I would like the server to allocate as little memory as possible per request, so that no GC pressure is generated and the application remains fast and responsive in high-usage scenarios.

**Example**

As a lot of tokens are known beforehand, we could evaluate them by comparing the memory fetched from the socket to the ASCII representation of the expected strings, meaning that we do not need to allocate memory for the string read from the socket just to compare it to a compile time constant. This would affect:

- HTTP verbs (GET, POST, ...)
- The HTTP protocol version
- Well-known header names

So basically everything that is represented by a special field in the `Request` implementation.

**Acceptance criteria**

- The existing tests still pass with the new implementation
- It has been shown that the number of allocations has been reduced (e.g. via dotMemory)
- It has been shown that there is no new hot path introduced by the implementation regarding the CPU usage (e.g. via dotTrace)
